### PR TITLE
remove kustomize commonLabels

### LINKED ISSUE #5887: 'commonLabels' is deprecated.
### Report

'commonLabels' is deprecated. Kustomizations should use 'labels' instead.

### Expected Behavior

When running `kustomize build .` on a modern version there is no deprecation warning.

### Actual Behavior

There is a deprecation warning:
```
# Warning: 'commonLabels' is deprecated. Please use 'labels' instead. Run 'kustomize edit fix' to update your Kustomization automatically.
```

### Steps to Reproduce the Problem

1. `cd config/default`
2. `kustomize build .` using kustomize 5.x

### Logs from KEDA operator

N/A

### KEDA Version

2.14.0

### Kubernetes Version

1.27

### Platform

Amazon Web Services

### Scaler Details

_No response_

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by therealdwright**
Signed-off-by: Daniel Wright <dan@dwright.io>

Remove `commonLabels` in favour of the more modern kustomize approach.

### Checklist

- [x] When introducing a new scaler, I agree with the [scaling governance policy](https://github.com/kedacore/governance/blob/main/SCALERS.md)
- [x] I have verified that my change is according to the [deprecations & breaking changes policy](https://github.com/kedacore/governance/blob/main/DEPRECATIONS.md)
- [x] Tests have been added
- [x] Changelog has been updated and is aligned with our [changelog requirements](https://github.com/kedacore/keda/blob/main/CONTRIBUTING.md#Changelog)
- [x] A PR is opened to update our Helm chart ([repo](https://github.com/kedacore/charts)) *(if applicable, ie. when deployment manifests are modified)*
- [x] A PR is opened to update the documentation on ([repo](https://github.com/kedacore/keda-docs)) *(if applicable)*
- [x] Commits are signed with Developer Certificate of Origin (DCO - [learn more](https://github.com/kedacore/keda/blob/main/CONTRIBUTING.md#developer-certificate-of-origin-signing-your-work))

Fixes #5887

### CODE REVIEW

File: config/rbac/kustomization.yaml [hunk 1]
```diff
@@ -1,8 +1,11 @@
...
 kind: Kustomization
-commonLabels:
-  app.kubernetes.io/name: keda-operator
-  app.kubernetes.io/part-of: keda-operator
+
+labels:
+  - pairs:
+      app.kubernetes.io/name: keda-operator
+      app.kubernetes.io/part-of: keda-operator
+    includeSelectors: true
```
Comment (zroubalik) [inline config/rbac/kustomization.yaml:8]: Why is this change needed?

Comment (therealdwright) [reply config/rbac/kustomization.yaml:8]: This change maintains feature parity with the current implementation of `commonLabels`. When you use `commonLabels,` it adds label selectors to all Kubernetes resources. This change addresses the deprecation warning without changing functionality/resulting manifests.

Comment (zroubalik) [reply config/rbac/kustomization.yaml:8]: Thanks!

**Review by zroubalik — COMMENTED** [review]
@therealdwright Thanks for the contribution, one last thing, could you please update Changelog (Unreleased/Other section) and link the issue there? We can proceed and merge this one then, Thanks!

**Comment by therealdwright** [comment]
> could you please update Changelog

No problems, that's done

**Review by zroubalik — APPROVED** [review]
LGTM
